### ISSUE TITLE
selftest.c doesn't working anymore with < 2.28.2 : MBEDTLS_SELF_TEST and MBEDTLS_TIMING_ALT

### ISSUE THREAD

**Issue by Kasimashi** [Mbed-TLS/mbedtls#6923]
### Summary

Having the MBEDTLS_SELF_TEST and MBEDTLS_TIMING_ALT flags activated I'm not able to link my project using mbedtls\programs\test\selftest.c

### System information

Mbed TLS version (number or commit id):  < 2.28.2
Operating system and version: All
Configuration (if not default, please attach `mbedtls_config.h`): 
Compiler and options (if you used a pre-built binary, please indicate how you obtained it): 
Additional environment information: 

### Expected behavior

Could be able to link the mbedtls library with selftest.c provided in example

### Actual behavior

Could not link the mbedtls library

### Steps to reproduce

I've an Error at the linking stage `	Error[Li005]: no definition for "mbedtls_timing_self_test" [referenced from mbedtls\programs\test\selftest.o]`

The reason is the changement on the commit :

Move the #endif of MBEDTLS_TIMING_ALT in timing.c

![image](https://user-images.githubusercontent.com/62567511/212296875-ccc0e570-8c97-4172-aba0-67dd48eae70b.png)

Could you please fix the bug or should we implement a selftest in timing_alt.c ?

**Comment by AndrzejKurek** [comment]
Hello @Kasimashi and thanks for all the details! I have raised a PR here: https://github.com/Mbed-TLS/mbedtls/pull/6931.
Let me know if this fixes the issue for you.

**Comment by Kasimashi** [comment]
Hello @AndrzejKurek .

Thanks you for the correction.

It fixed well the issue.

Thanks you ! ;)
You can close this issue and merge if it's also ok for you :)

Have a good day.
Kasimashi

**Comment by Kasimashi** [comment]
@paul-elliott-arm  I don't know if I have to close the issue could you please merge it before and then close this one ? I reopen it.

**Comment by AndrzejKurek** [comment]
> @paul-elliott-arm I don't know if I have to close the issue could you please merge it before and then close this one ? I reopen it.

We'll close the issue once [the PR](https://github.com/Mbed-TLS/mbedtls/pull/6931) is merged into development. It will be delivered with the next 2.28 release. Thanks for making sure that the fix works!

**Comment by tom-cosgrove-arm** [comment]
> We'll close the issue once the PR is merged into development

Minor niggle: [that PR](https://github.com/Mbed-TLS/mbedtls/pull/6931) will be merged in 2.28-LTS, hence why it will be in the next 2.28.x release. It won't be in development, as the code is different there.

**Comment by AndrzejKurek** [comment]
> > We'll close the issue once the PR is merged into development
> 
> Minor niggle: [that PR](https://github.com/Mbed-TLS/mbedtls/pull/6931) will be merged into 2.28-LTS, hence why it will be in the next 2.28.x release. It won't be in development, as the code is different there.
That's right, thanks for clarifying that :)

**Comment by Kasimashi** [comment]
@AndrzejKurek Pull request merged : I close the ticket.
